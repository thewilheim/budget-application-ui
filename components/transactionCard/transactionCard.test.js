
import TransactionCard from './index';
import { render, screen } from "@testing-library/react-native";
import renderer from "react-test-renderer";

jest.mock('@expo/vector-icons', () => ({
  Feather: '',
}));

const mockTransaction = {
    name: "Shoes",
    cost: "-$23.00",
    description: "Nike Shoes",
    date: "Aug 12"
}

describe('<TransactionCard />', () => {
  it('should render transaction card component', async () => {
    const tree = renderer.create(<TransactionCard name={mockTransaction.name} cost={mockTransaction.cost} description={mockTransaction.description} date={mockTransaction.date} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render transaction name', async () => {
    render(<TransactionCard name={mockTransaction.name} cost={mockTransaction.cost} description={mockTransaction.description} date={mockTransaction.date} />)
    expect(screen.getByText(mockTransaction.name)).toBeDefined()
  });
});