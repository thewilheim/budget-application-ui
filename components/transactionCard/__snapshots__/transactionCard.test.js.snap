// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders transaction card component 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "justifyContent": "space-between",
      },
      {
        "alignItems": "center",
      },
      {
        "width": "100%",
      },
      {
        "maxHeight": 80,
      },
      {
        "backgroundColor": "#fff",
      },
      {
        "flexDirection": "row",
      },
      {
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
      },
      {
        "shadowColor": "rgba(0, 0, 0, 0.1)",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
      },
      {
        "shadowColor": "rgba(55, 65, 81, 0.4)",
      },
      {
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
      },
      {
        "marginBottom": 4,
        "marginTop": 4,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          },
          {
            "backgroundColor": "#fdba74",
          },
          {
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
          },
          {
            "marginRight": 16,
          },
        ]
      }
    >
      <
        color="white"
        name="shopping-bag"
        size={24}
      />
    </View>
    <View>
      <Text
        style={
          [
            {
              "marginBottom": 4,
            },
            {
              "fontWeight": "700",
            },
          ]
        }
      >
        Shoes
      </Text>
      <Text
        style={
          [
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
            {
              "color": "#9ca3af",
            },
            {
              "fontWeight": "600",
            },
          ]
        }
      >
        Nike Shoes
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "textAlign": "center",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "marginBottom": 4,
          },
          {
            "fontWeight": "700",
          },
          {
            "color": "#f87171",
          },
          {
            "textAlign": "center",
          },
        ]
      }
    >
      -$23.00
    </Text>
    <Text
      style={
        [
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "color": "#9ca3af",
          },
          {
            "fontWeight": "600",
          },
        ]
      }
    >
      Aug 12
    </Text>
  </View>
</View>
`;
